pennylane/templates/subroutines/all_singles_doubles.py

```diff
@@ -0,0 +1,202 @@
+    in the subspace of two and four qubits, respectively.
+
+    The :class:`~.pennylane.SingleExcitation` operation performs a rotation
+    in the two-dimensional subspace :math:`\{\vert 10 \rangle, \vert 01 \rangle \}` of
+    the qubits ``[r, p]`` involved in the single excitation
+    :math:`\hat{c}_p^\dagger \hat{c}_r \vert \mathrm{HF} \rangle` of the initial state.
+    Similarly, the :class:`~.pennylane.DoubleExcitation` operations perform a rotation
+    in the subspace :math:`\{\vert 1100 \rangle, \vert 0011 \rangle \}` of
+    the qubits ``[s, r, q, p]`` that correspond to the double excitation
+    :math:`\hat{c}_p^\dagger \hat{c}_q^\dagger \hat{c}_r \hat{c}_s \vert \mathrm{HF} \rangle`.
+    The qubit indices ``s, r`` and ``q, p`` correspond, respectively, the occupied and unoccupied
+    spin-orbitals in the initial state, and :math:`c^\dagger` (:math:`c`) is the particle
+    creation (annihilation) operator. The total number of operations is determined by all
+    possible single- and double-excitations of the reference state.
+
+    For example, the quantum circuit for the case of two electrons and six qubits is sketched
+    in the figure below,
+
+    |
+
+    .. figure:: ../../_static/templates/subroutines/all_singles_doubles.png
+        :align: center
+        :width: 70%
+        :target: javascript:void(0);
+
+    |
+
+    In this case, we have four single- and double-excitations that preserves the total-spin
+    projection of the Hartree-Fock state. The :class:`~.pennylane.SingleExcitation` gates
+    :math:`G` act on the qubits ``[0, 2], [0, 4], [1, 3], [1, 5]`` as indicated by the
+    squares while the :class:`~.pennylane.DoubleExcitation` gates :math:`G^{(2)}` are applied
+    to the qubits ``[0, 1, 2, 3], [0, 1, 2, 5], [0, 1, 2, 4], [0, 1, 4, 5]``.
+
+    The resulting unitary conserves the number of particles and prepares the
+    :math:`n`-qubit system in a superposition of the initial HF state and
+    multiply-excited configurations.
+
+    Args:
+        weights (tensor_like): Size ``(len(singles) + len(doubles),)`` tensor containing the
+            angles :math:`\theta` entering the :class:`~.pennylane.SingleExcitation` and
+            :class:`~.pennylane.DoubleExcitation` operations. The indices of the qubits the
+            operations act on are generated with the :func:`~.excitations` function.
+        wires (Iterable): wires that the template acts on
+        hf_state (array[int]): Length ``len(wires)`` occupation-number vector representing the
+            HF state. ``hf_state`` is used to initialize the wires.
+        singles (Sequence[Sequence]): sequence of lists containing the wires indices ``[r, p]``
+        doubles (Sequence[Sequence]): sequence of lists containing the wires indices
+            ``[s, r, q, p]``
+
+    .. UsageDetails::
+
+        Notice that:
+
+        #. The number of wires has to be equal to the number of spin orbitals included in
```

Comment:
This and the following point would be obsolete if we pass only number of particles and qubits, so remember to remove

Reply:
This is connected with the comment below.